src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1645,11 +1645,11 @@ static void perform_stream_op_locked(void* stream_op,
   if (!grpc_core::IsCallTracerTransportFixEnabled()) {
     s->parent_call_tracer = ParentCallTracerIfSampled(s);
   }
-  // TODO(yashykt): Remove call_tracer field after transition to call v3. (See
-  // https://github.com/grpc/grpc/pull/38729 for more information.) On the
-  // client, the call attempt tracer will be available for use when the
-  // send_initial_metadata op arrives.
-  if (s->t->is_client && op->send_initial_metadata) {
+  // Some server filters populate CallTracerInterface in the context only after
+  // reading initial metadata. (Client-side population is done by
+  // client_channel filter.)
+  if (!t->is_client && !grpc_core::IsCallTracerInTransportEnabled() &&
```

Comment:
What is the rationale for adding the check !grpc_core::IsCallTracerInTransportEnabled()?